src/main/java/com/revlearn/team1/config/CorsConfig.java

```diff
@@ -0,0 +1,41 @@
+package com.revlearn.team1.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.cors.CorsConfiguration;
+import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
+
+import java.util.List;
+
+@Configuration
+public class CorsConfig {
+    @Bean
+    public UrlBasedCorsConfigurationSource corsConfigurationSource() {
+        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
+        source.registerCorsConfiguration("/**", corsConfiguration());
+        return source;
+    }
+
+    @Bean
+    public CorsConfiguration corsConfiguration() {
+        // TODO handle csrf properly
+
+        // TODO add any necessary EC2 URLs to this list
+        List<String> allowedOrigins = List.of(
```

Comment:
This is a lot of variations on the same thing. Would you want to use a regex here? IDK if that's an easy lift. if not, then ignore this

`https?:\/\/(localhost:(5173|8080)|(?:www\.)?revaturelearn\.com|api\.revaturelearn\.com)`